addons/hr_recruitment_integration_base/models/hr_job_post.py

```diff
@@ -0,0 +1,204 @@
+                platform=record.platform.name
+            )
+
+    def create(self, values):
+        posts = super().create(values)
+        posts._log_post_modifications()
+        return posts
+
+    def unlink(self):
+        for job_post in self:
+            job_post.sudo()._delete_post()
+        return super().unlink()
+
+    def _log_post_modifications(self, mode=None):
+        if mode is None:
+            mode = _('created')
+        for post in self:
+            post.message_post(
+                author_id=self.env.user.partner_id.id,
+                subject=_(
+                    'Job Post on %(platform)s has been modified',
+                    platform=post.platform.name
+                ),
+                body=Markup
+                    (_(
+                        '''
+                        <strong>Job Post on %(platform)s has been %(mode)s</strong> <br/>
+                        <strong>Job:</strong>  %(job)s\n<br/>
+                        <strong>Status:</strong>  %(status)s<br/>
+                        <strong>Status Message:</strong> %(status_message)s<br/>
+                        <strong>Contact Method:</strong> %(method)s<br/>
+                        <strong>Contact Point:</strong> %(vector)s<br/>
+                        <strong>Campaign Start Date:</strong> %(start_date)s<br/>
+                        <strong>Campaign End Date:</strong> %(end_date)s <br/>
+                        <strong>Company:</strong> %(company)s
+                        ''',
+                        platform=post.platform.name,
+                        mode=mode,
+                        job=post.job_id.name,
+                        status=dict(post._fields['status'].selection).get(post.status),
+                        status_message=post.status_message,
+                        method=dict(post._fields['apply_method'].selection).get(post.apply_method),
+                        vector=post.apply_vector,
+                        start_date=post.campaign_start_date,
+                        end_date=post.campaign_end_date or _('No Limit'),
+                        company=post.company_id.name,
+                    )
+                ),
+            )
+
+    def _end_finished_campaign(self):
+        posts_to_end = self.env['hr.job.post'].search([
+            ('status', 'not in', ['deleted', 'expired']),
+            ('campaign_end_date', '<=', fields.Date.today()),
+        ])
+        for post in posts_to_end:
+            post.action_delete_job_post()
+
+    def _start_new_campaign(self):
+        posts_to_start = self.env['hr.job.post'].search([
+            ('status', '=', 'pending'),
+            ('campaign_start_date', '<=', fields.Date.today()),
+            '|',
+                ('campaign_end_date', '>=', fields.Date.today()),
+                ('campaign_end_date', '=', False),
+        ])
+        for post in posts_to_start:
+            wizard = self.env['hr.recruitment.post.job.wizard'].create({
```

Comment:
As those wizards are gonna make to external calls and as you run this method in a cron job, you should commit each time to avoid making post infinitely each time the cron is stopped because of the large amount of posts to handle.

Reply:
Also, depending on the number of posts you could handle, maybe you should split ``posts_to_start`` into small batches, treat 1 (with 20 records for instance) and then call the cron to relaunch the process into another transaction using ``_trigger`` method.

Reply:
actually I was counting on the fact that the cron will time out on that case and retry but that's clearly cleaner